src/main/java/com/minecolonies/coremod/colony/buildings/modules/WorkerBuildingModule.java

```diff
@@ -0,0 +1,279 @@
+        this.jobEntry = entry;
+        this.primary = primary;
+        this.secondary = secondary;
+        this.canWorkingDuringRain = canWorkingDuringRain;
+        this.sizeLimit = sizeLimit;
+    }
+
+    @Override
+    public boolean assignCitizen(final ICitizenData citizen)
+    {
+        if (citizen.getWorkBuilding() != null)
+        {
+            citizen.getWorkBuilding().getFirstOptionalModuleOccurance(WorkerBuildingModule.class).ifPresent(m -> m.removeCitizen(citizen));
+        }
+
+        if (!super.assignCitizen(citizen))
+        {
+            Log.getLogger().warn("Unable to assign citizen:" + citizen.getName() + " to building:" + building.getSchematicName() + " jobname:" + getJobDisplayName());
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public void deserializeNBT(final CompoundNBT compound)
+    {
+        if (compound.getAllKeys().contains(TAG_WORKER))
+        {
+            final ListNBT workersTagList = compound.getList(TAG_WORKER, Constants.NBT.TAG_COMPOUND);
+            for (int i = 0; i < workersTagList.size(); ++i)
+            {
+                final ICitizenData data = building.getColony().getCitizenManager().getCivilian(workersTagList.getCompound(i).getInt(TAG_WORKER_ID));
+                if (data != null && data.getJob() != null && data.getJob().getJobRegistryEntry().equals(jobEntry))
+                {
+                    assignCitizen(data);
+                }
+            }
+            this.hiringMode = HiringMode.values()[compound.getInt(TAG_HIRING_MODE)];
+        }
+        else if (compound.contains(jobEntry.getKey().toString()))
+        {
+            final CompoundNBT jobCompound = compound.getCompound(jobEntry.getKey().toString());
+            final int[] residentIds = jobCompound.getIntArray(TAG_WORKING_RESIDENTS);
+            for (final int citizenId : residentIds)
+            {
+                final ICitizenData citizen = building.getColony().getCitizenManager().getCivilian(citizenId);
+                if (citizen != null)
+                {
+                    assignCitizen(citizen);
+                }
+            }
+            this.hiringMode = HiringMode.values()[jobCompound.getInt(TAG_HIRING_MODE)];
+        }
+    }
+
+    @Override
+    public void onColonyTick(@NotNull final IColony colony)
+    {
+        // If we have no active worker, grab one from the Colony
+        if (!isFull() && ((building.getBuildingLevel() > 0 && building.isBuilt()) || building instanceof BuildingBuilder)
+              && (this.hiringMode == HiringMode.DEFAULT && !building.getColony().isManualHiring() || this.hiringMode == HiringMode.AUTO))
+        {
+            final ICitizenData joblessCitizen = colony.getCitizenManager().getJoblessCitizen();
+            if (joblessCitizen != null)
+            {
+                assignCitizen(joblessCitizen);
+            }
+        }
+    }
+
+    @Override
+    public void serializeNBT(final CompoundNBT compound)
```

Comment:
No, there is none in any super class.

Reply:
ah should probably move the saving of the citizen list into the assignment module then and serialize/deserialize it there

Reply:
No, on purpose I didn't because we want to properly divide the "housing assignment" from the "living assignment". That is especially important for backwards compat.